docs/en/ingest-management/fleet/fleet-server-secrets.asciidoc

```diff
@@ -0,0 +1,265 @@
+
+[[secret-files-guide]]
+== Secret files guide
+
+This guide provides step-by-step examples with best practices on how to deploy secret files directly on a host or through the Kubernetes secrets engine.
+
+[[secret-filesystem]]
+=== Secrets on filesystem
+
+Secret files can be provisioned as plain text files directly on filesystems and referenced or passed through {agent}.
+
+We recommend the following steps in order to improve security.
+
+==== File Permissions
+
+File permissions should not allow for global read permissions.
+
+On MacOS and Linux, file ownership and file permissions can be set with the `chown` and `chmod` commands, respectively.
+{fleet-server} runs as the `root` user on MacOS and Linux, so given a file named `mySecret`, it can be altered with:
+[source,sh]
+----
+sudo chown root:root mySecret # set the user:group to root
+sudo chmod 0600 mySecret      # set only the read/write permission flags for the user, clear group and global permissions.
+----
+
+On Windows, you can use `icacls` to alter the ACL list associated with the file:
+[source,powershell]
+----
+Write-Output -NoNewline SECRET > mySecret          # Create the file mySecret with the contents SECRET
+icacls .\mySecret /inheritance:d                   # Remove inherited permissions from file
+icacls .\mySecret /remove:g BUILTIN\Administrators # Remove Administrators group permissions
+icacls .\mySecret /remove:g $env:UserName          # Remove current user's permissions
+----
+
+==== Temporary filesystem
+
+A temporary filesystem (in RAM) can be used to hold secret files in order to improve security.
+These types of filesystems are normally not included in backups and cleared if the host is reset.
+However, the filesystem and secret files need to be reprovisioned with every reset.
+
+On Linux `mount` can be used with the `tmpfs` filesystem to create a temporary filesystem in RAM:
+[source,sh]
+----
+mount -o size=1G -t tmpfs none /mnt/fleet-server-secrets
+----
+
+On MacOS a combination of `diskutil` and `hdiutil` can be used to create a RAM disk:
+[source,sh]
+----
+diskutil erasevolume HFS+ 'RAM Disk' `hdiutil attach -nobrowse -nomount ram://2097152`
+----
+
+For Windows systems, there are no built-in options to create a RAM disk.
+If this is required, a third party program is needed.
+
+==== Example
+
+Here is a step by step guide for provisioning a service token on a Linux system:
+[source,sh]
+----
+sudo mkdir -p /mnt/fleet-server-secrets
+sudo mount -o size=1G -t tmpfs none /mnt/fleet-server-secrets
+echo -n MY-SERVICE-TOKEN > /mnt/fleet-server-secrets/service-token
+sudo chown root:root /mnt/fleet-server-secrets/service-token
+sudo chmod 0600 /mnt/fleet-server-secrets/service-token
+----
+
+NOTE: The `-n` flag is used with `echo` to prevent a newline character from being appended at the end of the secret. Care should be taken so that the secret file does not contain the trailing newline character.
```

Comment:
```suggestion
NOTE: The `-n` flag is used with `echo` to prevent a newline character from being appended at the end of the secret. Be sure that the secret file does not contain the trailing newline character.
```